vault/acl.go

```diff
@@ -47,25 +50,58 @@ func NewACL(policies []*Policy) (*ACL, error) {
 			// Check for an existing policy
 			raw, ok := tree.Get(pc.Prefix)
 			if !ok {
-				tree.Insert(pc.Prefix, pc.CapabilitiesBitmap)
+				tree.Insert(pc.Prefix, pc.Permissions)
 				continue
 			}
-			existing := raw.(uint32)
+
+			// these are the ones already in the tree
+			existingPerms := raw.(*Permissions)
```

Comment:
Now that I think more about it, this makes more sense. I don't think it'll panic since the object we build with the new binary will only have `*Permissions` in the tree.

Ignore the comment